packages/backend/src/modules/tracked-txs/modules/l2-costs/L2CostsUpdater.ts

```diff
@@ -0,0 +1,86 @@
+import { Logger } from '@l2beat/backend-tools'
+import { assertUnreachable, UnixTime } from '@l2beat/shared-pure'
+import { Knex } from 'knex'
+
+import { ViemRpcClient } from '../../../../peripherals/viem-rpc-client/ViemRpcClient'
+import { TrackedTxResult } from '../../types/model'
+import { TrackedTxId } from '../../types/TrackedTxId'
+import { TxUpdaterInterface } from '../../types/TxUpdaterInterface'
+import {
+  L2CostsRecord,
+  L2CostsRepository,
+} from './repositories/L2CostsRepository'
+
+export class L2CostsUpdater implements TxUpdaterInterface {
+  constructor(
+    private readonly l2CostsRepository: L2CostsRepository,
+    private readonly rpcClient: ViemRpcClient,
+    private readonly logger: Logger,
+  ) {}
+
+  async update(transactions: TrackedTxResult[], knexTx?: Knex.Transaction) {
+    if (transactions.length === 0) {
+      this.logger.debug('[L2Costs]: Update skipped')
+      return
+    }
+
+    const transformedTransactions =
+      await this.addDetailsTransactionsAndTransform(transactions)
+    await this.l2CostsRepository.addMany(transformedTransactions, knexTx)
+  }
+
+  async deleteAfter(
+    id: TrackedTxId,
+    untilTimestamp: UnixTime,
+    knexTrx: Knex.Transaction,
+  ) {
+    await this.l2CostsRepository.deleteAfter(id, untilTimestamp, knexTrx)
+  }
+
+  async addDetailsTransactionsAndTransform(
+    transactions: TrackedTxResult[],
+  ): Promise<L2CostsRecord[]> {
+    const promises = transactions.map(async (tx) => {
+      if (
+        tx.transactionType === 0 ||
+        tx.transactionType === 1 ||
+        tx.transactionType === 2
+      ) {
+        return {
+          timestamp: tx.blockTimestamp,
+          txHash: tx.hash,
+          trackedTxId: tx.use.id,
+          data: {
+            type: tx.transactionType,
+            gasUsed: tx.receiptGasUsed,
+            gasPrice: tx.gasPrice,
+            calldataLength: tx.dataLength,
+            calldataGasUsed: tx.calldataGasUsed,
+          },
+        }
+      } else if (tx.transactionType === 3) {
```

Comment:
GRRRRRRRR

Reply:
1 beer for using else if

Reply:
change that big if to switch case statement

Reply:
[d5037a5](https://github.com/l2beat/l2beat/pull/3007/commits/d5037a5221a1d1061d149f970880363a8ccbb668)